### ISSUE TITLE
Adding watch paths to git config doesn’t automatically trigger commits when those paths are updated

### ISSUE THREAD

**Issue by fitzage** [statamic/cms#2400]
## Bug Description
I added a folder to the paths array in git.php
What I expected to happen:
- When Statamic made changes to this folder, I expected the changes to be committed to git.
What happened:
- The changes are not committed to git, but just show as staged for commit.
- I can get them committed if I manually run `php please git:commit`
- I can get them committed if I do some other function that triggers a commit (like adding or editing an asset).

## Extra Detail
- The folder in question is the `storage/statamic/revisions` folder, which I set up to no longer be ignored by git
- Moving this folder out to the top level and changing the location in `config/revisions.php` results in the same behavior
- STATAMIC_GIT_AUTOMATIC is set to true (and is working as expected for other changes)
- So statamic now knows to commit these changes to git, but it doesn’t automatically trigger a commit when these files are updated

## Environment

Statamic 3.0.7 Pro
Laravel 7.25.0
PHP 7.3.17
statamic/seo-pro 2.0.11
Fresh install from `statamic/statamic`, migrated from v2

**Comment by jasonvarga** [comment]
I think the real issue is that Git isn't hooked up to the revisions stuff.

The list of directories aren't being 'watched' for changes. Like you mentioned, there needs to be some action that the git integration is looking for. Like an entry being saved. We need to hook it up to revisions being created.

Hasn't happened yet because honestly we didn't think people would want to track revisions in git. They are gitignored by default.

**Comment by fitzage** [comment]
Right. So the documentation seems to indicate that just adding paths to the git.php config makes this happen, but that’s not really the case.

**Comment by fitzage** [comment]
I still need a solution for this before I can use the revisions feature.

**Comment by fitzage** [comment]
Adding the note that it turns out there isn’t actually an event for revisions at this point, so there’s nothing to tie into yet.
